### PR TITLE
[security] Don't generate 1024-bit RSA keys

### PR DESCRIPTION
**Pull request by alex**
## Contributor Checklist:

* [x] There is an associated ticket in Trac. Create a new one at https://twistedmatrix.com/trac/newticket
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [x] I have updated the automated tests.

### CODE REVIEW

**Comment by alex** [comment]
Test failures are due to the bug being fixed in #1014

**Comment by alex** [comment]
https://twistedmatrix.com/trac/ticket/9453

File: src/twisted/newsfragments/9453.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+twisted.internet._sslverify.KeyPair.generate: No longer generate 1024-bit RSA keys by default. Anyone who generated a key with this method using the default value should move to replace it immediately.
```
Comment (exarkun) [inline src/twisted/newsfragments/9453.bugfix]: The public name of the API is twisted.internet.ssl.KeyPair.

Comment (exarkun) [reply src/twisted/newsfragments/9453.bugfix]: Also the style guideline for news fragments suggests this should be more like:

    twisted.internet.ssl.KeyPair.generate now defaults to generating a 2048 bit key.

I doubt that this is an effective place to provide "immediate" security action guidance.

**Review by markrwilliams — APPROVED** [review]
[1024 is not enough](https://wiki.mozilla.org/CA:MD5and1024).  Thanks!

**Comment by alex** [comment]
@markrwilliams do you need anything from me on this?

**Comment by markrwilliams** [comment]
@alex Sorry for my terse message.  You don't need to do anything more - this only needs successful builds from Appveyor and Travis, both of which failed for unrelated reasons:

https://ci.appveyor.com/project/adiroiban/twisted/build/trunk-5076/job/kh8d8chxucxo3ni8#L18437
https://travis-ci.org/twisted/twisted/jobs/393945255#L4491

**Comment by alex** [comment]
@markrwilliams thanks! they look to be green now.

**Review by mstojcevich — APPROVED** [review]

**Comment by twm** [comment]
`admin/pr_as_branch 1015 9453 1024-is-not-enough-bits`

**Pull request merged**